### ISSUE TITLE
Documentation is missing explanation of how to write custom loss functions for the keras api

### ISSUE THREAD

**Issue by esslushy** [tensorflow/tensorflow#31202]
## URL(s) with the issue:

https://www.tensorflow.org/beta/guide/keras/training_and_evaluation#specifying_a_loss_metrics_and_an_optimizer

## Description of issue (what needs changing):

Documentation is missing how to write custom loss functions and how to use the keras.losses.Loss class to write a loss function that takes more inputs than just y_true and y_pred despite the portion saying it will show how to write custom losses and metrics. Currently it shows how to write custom metrics and in layer loss only.

### Clear description

People can use this method to implement loss function not implemented in tensorflow or their own variations of loss functions. It also allows ports of loss functions without a keras.losses exposure such as tf.nn.weighted_cross_entropy_with_logits to be implemented in a model.compile use case without the need to write a custom training loop.

### Correct links

N/A

### Parameters defined

N/A

### Returns defined

N/A

### Raises listed and defined

N/A

### Usage example

There is no documentation, but a good example would be porting a loss function from tensorflow.nn to be used in a class extending keras.losses.Loss and showing how to use a user written function without the keras.losses.Loss class.

### Request visuals, if applicable

No visuals are there, but a portion of code similar to the custom metric code shown would be a good visual. 

A custom loss function I wrote in python follows:
```
class WeightedBinaryCrossEntropy(keras.losses.Loss):
    """
    pos_weight: Scalars the effec on loss by the positive class by whatever is passed into it.
    weight: Scalars all the loss. Can be used to increase scalar of negative weight only by passing 1/weight to pos_weight. 
            To affect pos_weight even more after this multiply in the other scalar you had in mind for it
    """
    def __init__(self, pos_weight, weight, from_logits=False, reduction=keras.losses.Reduction.AUTO, name='weighted_binary_crossentropy'):
        super(WeightedBinaryCrossEntropy, self).__init__(reduction=reduction, name=name)
        self.pos_weight = pos_weight
        self.weight = weight
        self.from_logits = from_logits

    def call(self, y_true, y_pred):
        if not self.from_logits:
            with tf.name_scope('Weighted_Cross_Entropy'):
                # Manually calculated the weighted cross entropy. Formula is qz * -log(sigmoid(x)) + (1 - z) * -log(1 - sigmoid(x)) where z are labels, x is logits, and q is the weight.
                # Since the values passed are from sigmoid (assumably in this case) sigmoid(x) will be replaces with y_pred
                x_1 = y_true * self.pos_weight * -tf.math.log(y_pred + 1e-6) # qz * -log(sigmoid(x)) 1e-6 is added as an epsilon to stop passing a zero into the log
                x_2 = (1 - y_true) * -tf.math.log(1 - y_pred + 1e-6) # (1 - z) * -log(1 - sigmoid(x)). Epsilon is added to prevent passing a zero into the log
                return tf.add(x_1, x_2) * self.weight # Must be negative as it is maximized when passed to optimizers
        # Use built in function
        return tf.nn.weighted_cross_entropy_with_logits(y_true, y_pred, self.pos_weight) * self.weight
```

### Submit a pull request?

Are you planning to also submit a pull request to fix the issue? See the docs
contributor guide: https://www.tensorflow.org/community/contribute/docs and the
docs style guide: https://www.tensorflow.org/community/contribute/docs_style

**Comment by jvishnuvardhan** [comment]
@esslushy Are you planning to submit a PR to improve the documentation with examples of custom loss function? Thanks!

**Comment by esslushy** [comment]
@jvishnuvardhan Sure I can do that. I just was unsure if I should in case I misunderstood how the losses api was suppose to be shown. I will work on a PR. Thanks for the response.

**Comment by jvishnuvardhan** [comment]
@esslushy I know you already know what you need to do. There was a recent [issue](https://github.com/tensorflow/tensorflow/issues/30711) regarding custom metric which might help you. Thanks!

**Comment by esslushy** [comment]
@jvishnuvardhan thanks for the issue regarding the metrics and for responding so quickly. I have submit a [pull request](https://github.com/tensorflow/docs/pull/892) for the change. Thanks for all the help. Have a great day.

**Comment by jvishnuvardhan** [comment]
@esslushy Automatically closing this out since I understand it to be resolved by the https://github.com/tensorflow/docs/pull/892 (merged already), but please let me know if I'm mistaken.Thanks!
